File: python/paypal.py
import requests

# Tus credenciales de PayPal
CLIENT = 'Ac49PuTgBmuyf-VlmLq4Axf5BX3F5O3uO81kbsO_GTBrErCfrMeE9DwAD37LapgPeV6R2cn3ik7sjvkP'
SECRET = 'EFbC49mwdaABS-iIl59jxczdL1lUJJZKe2WzjvOJXLqPTZH9judvhZfjRVU_pkm9R5jGTgrQVRPl8nrm'
PAYPAL_API = 'https://api-m.sandbox.paypal.com'

# Los datos de tu orden
body = {
    'intent': 'CAPTURE',
    'purchase_units': [{
        'amount': {
            'currency_code': 'USD',
            'value': '100'
        }
    }],
    'application_context': {
        'brand_name': 'MotorsSolution',
        'landing_page': 'NO_PREFERENCE',
        'user_action': 'PAY_NOW',
        'return_url': 'http://localhost:3000/execute-payment',
        'cancel_url': 'http://localhost:3000/cancel-payment'
    }
}

# Hacer la solicitud POST a la API de PayPal
response = requests.post(f'{PAYPAL_API}/v2/checkout/orders', auth=(CLIENT, SECRET), json=body)

# Extraer el enlace de aprobación de la respuesta
data = response.json()
paypal_url = next(link['href'] for link in data['links'] if link['rel'] == 'approve')

print(paypal_url)

File: js/session/paypal.js
// paypal.js
const axios = require('axios');
const dotenv = require("dotenv");
dotenv.config({ path: "env/.env" });

const CLIENT = process.env.CLIENT;
const SECRET = process.env.SECRET;
const PAYPAL_API = process.env.PAYPAL_API;

// Los datos de tu orden
const body = {
    intent: 'CAPTURE',
    purchase_units: [{
        amount: {
            currency_code: 'USD',
            value: '10'
        }
    }],
    application_context: {
        brand_name: 'MotorsSolution',
        landing_page: 'NO_PREFERENCE',
        user_action: 'PAY_NOW',
        return_url: `https://motorssolution.onrender.com/execute-payment`,
        cancel_url: `https://motorssolution.onrender.com/cancel-payment`
//      return_url: `HTTP://localhost:3000/execute-payment`,
//      cancel_url: `HTTP://localhost:3000/cancel-payment`
    }
};

function getPaypalUrl() {
  return axios.post(`${PAYPAL_API}/v2/checkout/orders`, body, {
    auth: {
        username: CLIENT,
        password: SECRET
    }
  }).then(response => {
    // Extraer el enlace de aprobación de la respuesta
    return response.data.links.find(link => link.rel === 'approve').href;
  });
}

module.exports = getPaypalUrl;
